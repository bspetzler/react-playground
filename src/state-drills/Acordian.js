import React, {Component} from 'react';

export default class Acordian extends Component {
    state ={
        currentIndex: null
    }

    static defaultProps = {
        sections: []
    }

    handelButtonClick = (index) =>{
        this.setState({ currentIndex: index });
        //console.log(this.state.currentIndex)
    }

    renderContent(sec, index, showing){
        //console.log(showing)
        return (
                <li key={index}>
                    <button onClick ={() => this.handelButtonClick(index)}> {sec.title} </button>
                    {(showing === index) && <p>{sec.content}</p>} {/*Conditional rendering */}
                </li>
          )
    }

    render(){
        const showing = this.state.currentIndex
        return (
            <ul className='content'>
                {this.props.sections.map((sec, index) => 
                    this.renderContent(sec, index, showing)
                )}
            </ul>
        )
    }
}