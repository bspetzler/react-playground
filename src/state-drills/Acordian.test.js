import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Acordian from './Acordian';

const sect = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

  describe (`Acordian Component`, () => {
      it('renders when the page loads', () => {
          const wrapper = shallow(<Acordian />)
          expect(toJson(wrapper)).toMatchSnapshot()
      })

      it('renders no content to start', () => {
          const wrapper = shallow(<Acordian sections={sect} />)
          expect(toJson(wrapper)).toMatchSnapshot()
      })
      
      it('opens sections correctly', () => {
          const wrapper = shallow(<Acordian sections={sect} />)
          wrapper.find('button').at(0).simulate('click')
          wrapper.find('button').at(1).simulate('click')
          wrapper.find('button').at(2).simulate('click')
          expect(toJson(wrapper)).toMatchSnapshot()
      })

  })