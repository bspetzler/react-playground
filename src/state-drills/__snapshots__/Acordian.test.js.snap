// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Acordian Component opens sections 1`] = `
<ul
  className="content"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
       
      Section 1
       
    </button>
     
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
       
      Section 2
       
    </button>
     
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
       
      Section 3
       
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
     
  </li>
</ul>
`;

exports[`Acordian Component opens sections correctly 1`] = `
<ul
  className="content"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
       
      Section 1
       
    </button>
     
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
       
      Section 2
       
    </button>
     
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
       
      Section 3
       
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
     
  </li>
</ul>
`;

exports[`Acordian Component renders no content to start 1`] = `
<ul
  className="content"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
       
      Section 1
       
    </button>
     
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
       
      Section 2
       
    </button>
     
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
       
      Section 3
       
    </button>
     
  </li>
</ul>
`;

exports[`Acordian Component renders when the page loads 1`] = `
<ul
  className="content"
/>
`;
